test/module/s3.spec.js

```diff
@@ -188,15 +197,71 @@ describe('Testing s3 Util', { useNock: true, timestamp: 1569876020 }, () => {
   });
 
   it('Testing "listObjects" with "StopAfter" no items returned', async () => {
-    const result = await aws.s3.listObjects({
+    const result = await S3().listObjects({
       bucket,
       stopAfter: '2020-03-30T15:10:00.000Z'
     });
     expect(result).to.deep.equal([]);
   });
 
   it('Testing "decodeKey"', () => {
-    const result = aws.s3.decodeKey('2018-10-25T20%3A55%3A00.000Z/Collection+Viewed.json.gz');
+    const result = S3().decodeKey('2018-10-25T20%3A55%3A00.000Z/Collection+Viewed.json.gz');
     expect(result).to.equal('2018-10-25T20:55:00.000Z/Collection Viewed.json.gz');
   });
+
+  it('Testing error rate exceeds retry count', async ({ capture, recorder }) => {
+    const s3 = S3({
+      logger: console,
+      backoffFunction: () => 0,
+      maxRetries: 1
+    });
+    const error = await capture(() => s3.putGzipObject({
+      bucket,
+      key,
+      data: JSON.stringify({ data: 'data' })
+    }));
+    expect(error).to.deep.contain({
+      statusCode: 400
+    });
+    expect(recorder.get()).to.deep.equal([
+      'Failed to submit (some) s3:putObject(s)\n'
+      + 'Retrying: [{"action":"s3:putObject","opts":{"ContentType":"application/json","ContentEncoding":"gzip",'
+      + '"Bucket":"test-bucket-name","Key":"key","Body":{"type":"Buffer","data":[31,139,8,0,0,0,0,0,2,3,171,86,74,'
+      + '73,44,73,84,178,130,80,181,0,185,30,67,221,15,0,0,0]}}}]'
+    ]);
+  });
+
+  it('Testing error rate until fatal error', async ({ capture, recorder }) => {
+    const s3 = S3({
+      logger: console,
+      backoffFunction: () => 0,
+      maxRetries: 2
+    });
+    const error = await capture(() => s3.putGzipObject({
+      bucket,
+      key,
+      data: JSON.stringify({ data: 'data' })
+    }));
+    expect(error).to.deep.contain({
+      statusCode: 500
+    });
+    expect(recorder.get()).to.deep.equal([
+      'Failed to submit (some) s3:putObject(s)\n'
+      + 'Retrying: [{"action":"s3:putObject","opts":{"ContentType":"application/json","ContentEncoding":"gzip",'
+      + '"Bucket":"test-bucket-name","Key":"key","Body":{"type":"Buffer","data":[31,139,8,0,0,0,0,0,2,3,171,86,74,'
+      + '73,44,73,84,178,130,80,181,0,185,30,67,221,15,0,0,0]}}}]'
+    ]);
+  });
```

Comment:
This test is taking ~(780ms) no matter the `backoffFunction` is zero.
Probably is because the two `aws` request